tm2/pkg/crypto/keys/client/deprecated.go

```diff
@@ -0,0 +1,19 @@
+package client
+
+import "github.com/gnolang/gno/tm2/pkg/commands"
+
+// Deprecated: NewRootCmd is deprecated and has been moved in
+// `gno.land/pkg/keycli`. Please update your code to use the new package. This
+// function now only serves as a placeholder and will panic if called to
+// encourage migration.
+func NewRootCmd(io commands.IO) *commands.Command {
+	panic("NewRootCmd: has been deprecated, use `gno.land/pkg/keycli` instead")
+}
```

Comment:
Do you think we should drop this entirely instead of leaving the API?

Reply:
For me, leaving a portion of the API here will ultimately lead to either duplicate code or the necessity to expose parts of the client that were not intended to be exposed. 
Personally, I would prefer relocating this specific client to `gno.land` (so it's not removed, just moved elsewhere) and exposing a more generic library here. This library could serve as a versatile tool that can power `gnokey` or any other tool, not limited to just CLI tools.

Reply:
We should avoid using deprecated methods and instead opt for a compilation-time approach.

Reply:
Something like this: https://github.com/quic-go/quic-go/pull/3364/files

Reply:
@gfanton attempted it and shared a screenshot of the outcome, which unfortunately didn't meet our expectations.

Let's proceed by removing the .go file and retaining only the README.md file; this approach should suffice.

Reply:
It kind of works, but the issue is that it will leave a permanent error in the code, not just on import. This is especially problematic for tools like `golangci-lint` and `gopls`, so it's not suitable for us.